infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerRuntimeContext.java

```diff
@@ -77,11 +104,34 @@ public URI getOutputChannel() throws InfrastructureException {
     }
 
     @Override
-    public InternalRuntime getRuntime() {
-        return dockerRuntimeFactory.createRuntime(this,
-                                                  devMachineName,
-                                                  orderedContainers,
-                                                  dockerEnvironment,
-                                                  identity);
+    public DockerInternalRuntime getRuntime() throws InfrastructureException {
+        List<ContainerInfo> runningContainers = containers.find(identity);
```

Comment:
Once i change the code it won't be as costly as it is, it think that caching will be a nice improvement, but it also brings additional complexity, i'm not sure i can handle it in this pr.